apps/journeys-admin/src/components/FramePortal/FramePortal.tsx

```diff
@@ -0,0 +1,94 @@
+import { createPortal } from 'react-dom'
+import createCache from '@emotion/cache'
+import { CacheProvider } from '@emotion/react'
+import {
+  DetailedHTMLProps,
+  IframeHTMLAttributes,
+  memo,
+  ReactElement,
+  ReactNode,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef
+} from 'react'
+import { styled } from '@mui/material'
+import { StylesProvider } from '@mui/styles'
+
+interface ContentProps {
+  children: ReactNode
+  document: Document
+}
+
+function Content({ children, document }: ContentProps): ReactElement {
+  const cache = useMemo(
+    () =>
+      createCache({
+        key: 'iframe',
+        container: document.head,
+        prepend: true
+      }),
+    [document]
+  )
+
+  return (
+    <StylesProvider>
+      <CacheProvider value={cache}>{children}</CacheProvider>
+    </StylesProvider>
+  )
+}
+
+const StyledFrame = styled('iframe')(() => ({
+  border: 0
+}))
+
+interface FrameProps
+  extends Omit<
+    DetailedHTMLProps<
+      IframeHTMLAttributes<HTMLIFrameElement>,
+      HTMLIFrameElement
+    >,
+    'css'
+  > {
+  children: ReactNode
+}
+
+export const FramePortal = memo(function FramePortal({
+  children,
+  ...other
+}: FrameProps): ReactElement {
+  const frameRef = useRef<HTMLIFrameElement>(null)
+  // If we portal content into the iframe before the load event then that
+  // content is dropped in firefox.
+  const [iframeLoaded, onLoad] = useReducer(() => true, false)
```

Comment:
```suggestion
  const [iframeLoaded, dispatch] = useReducer(() => true, false)
```
or use `dispatchLoad`? Keeping to React terminology makes it a bit more familiar to read + on___ comes with all sorts of other connotations.

Reply:
done